wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/WorkspaceValidator.java

```diff
@@ -166,6 +176,35 @@ private void validateLongAttribute(
     }
   }
 
+  /**
+   * Check that workspace has either plugins defined in attributes (Che 7) or installers defined in
+   * machines (Che 6).
+   *
+   * @throws ValidationException
```

Comment:
Please add few words when this exception can be thrown

Reply:
Fixed